### ISSUE TITLE
Superfluous use of `partprobe`

### ISSUE THREAD

**Issue by codefiles** [archlinux/archinstall#2211]
Pertinent `partprobe` manual information: [1]

> **partprobe** is a program that informs the operating system kernel of partition table changes.

From pyparted, `Disk.commit()` also "informs the operating system [kernel] of the changes" to a partition table [2] therefore calling `partprobe` after `DeviceHandler._perform_partitioning()` is unnecessary.

https://github.com/archlinux/archinstall/blob/f6106324eb71905df9b832eb973a08a1f45fa5af/archinstall/lib/disk/device_handler.py#L327-L328

https://github.com/archlinux/archinstall/blob/f6106324eb71905df9b832eb973a08a1f45fa5af/archinstall/lib/disk/device_handler.py#L366-L367

https://github.com/archlinux/archinstall/blob/f6106324eb71905df9b832eb973a08a1f45fa5af/archinstall/lib/disk/device_handler.py#L523-L529

A `partprobe` that is performed after formatting the partitions is unnecessary because formatting the partitions does not change the partition table.

https://github.com/archlinux/archinstall/blob/f6106324eb71905df9b832eb973a08a1f45fa5af/archinstall/lib/disk/device_handler.py#L295-L305

https://github.com/archlinux/archinstall/blob/f6106324eb71905df9b832eb973a08a1f45fa5af/archinstall/lib/disk/device_handler.py#L377-L389

Encrypting a partition does not change the partition table.

https://github.com/archlinux/archinstall/blob/f6106324eb71905df9b832eb973a08a1f45fa5af/archinstall/lib/luks.py#L165

[1] https://man.archlinux.org/man/partprobe.8
[2] https://github.com/dcantrell/pyparted/blob/d5870bb7a5d512c17c92a4e4fe7e9a8e7c8c3528/src/parted/disk.py#L202-L205

**Comment by svartkanin** [comment]
Hmm I remember there was a reason we put the partprobe in place after partitioning, I can't fully remember now but I think there was an issue without it not loading the lsblk info correctly

**Comment by codefiles** [comment]
Regardless of whatever the intent was in the past, no evidence has been provided to justify any of the uses of partprobe outlined in the issue.

**Comment by qlyoung** [comment]
This reintroduced #1759

**Comment by codefiles** [comment]
I doubt that but I could be wrong, logs would help. This type of issue appears to occur when partitions get created with invalid disk geometry (see #2210). I discovered another issue that will cause problems with partitioning and will raise an issue in a moment.

**Comment by qlyoung** [comment]
> I doubt that but I could be wrong

I readded a single call to `partprobe()` as part of the retry logic after calling `lsblk` fails and it worked - precisely the same as the original patch that resolved #1759.

>  logs would help

Getting logs for you would be rather inconvenient.

> This type of issue appears to occur when partitions get created with invalid disk geometry (see https://github.com/archlinux/archinstall/pull/2210).

My geometry seems valid enough since the system installed with `archinstall` after my workaround subsequently booted and works perfectly. I checked manually and it seems fine. It's nothing complicated, just an ESP and a primary.

If I submit my patch, will you nack it? Just trying to gauge up front how much effort it's going to take for me to fix this.
